content/blog/nanoparquet-0-4-0/index.Rmd

```diff
@@ -0,0 +1,230 @@
+---
+output: hugodown::hugo_document
+
+slug: nanoparquet-0-4-0
+title: nanoparquet 0.4.0
+date: 2025-01-28
+author: Gábor Csárdi
+description: >
+    nanoparquet 0.4.0 comes with a new and much faster `read_parquet()`,
+    configurable type mappings in `write_parquet()`, and a new
+    `append_parquet()`.
+
+photo:
+  url: https://www.pexels.com/photo/person-running-in-the-hallway-796545/
+  author: Michael Foster
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [package]
+tags: [parquet]
+---
+
+<!--
+TODO:
+* [x] Look over / edit the post's title in the yaml
+* [x] Edit (or delete) the description; note this appears in the Twitter card
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [x] `usethis::use_tidy_thanks()`
+-->
+
+We're thrilled to announce the release of [nanoparquet](
+  https://nanoparquet.r-lib.org/) 0.4.0. nanoparquet is an R package that
+reads and writes Parquet files.
+
+You can install it from CRAN with:
+
+```r
+install.packages("nanoparquet")
+```
+
+This blog post will show the most important new features of nanoparquet
+0.4.0: You can see a full list of changes in the [release notes]({ TODO }).
+
+## Brand new `read_parquet()`
+
+nanoparquet 0.4.0 comes with a completely rewritten Parquet reader.
+The new version has an architecture that is easier to embed into R, and
+facilitates great features, like `append_parquet()` and the new
+`col_select` argument. (More to come!) The new reader is also much faster,
+see the "Benchmarks" chapter.
+
+## Read a subset of columns
+
+`read_parquet()` now has a new argument called `col_select`, that lets you
+read a subset of the columns from the Parquet file. Unlike for row oriented
+file formats like CSV, this means that the reader never needs to touch the
+columns that are not needed for analysis. Reading a handful of columns only
+always takes the same amount of time, no matter how many more columns the
+Parquet file might have!
```

Comment:
I get what you're trying to say here but "same amount of time as what?". So maybe just something like "This makes it much faster to read only the columns you care about"

Reply:
Yeah, I rewrite this, I think it is better now.